pkg/vex/vex.go

```diff
@@ -145,7 +145,7 @@ func (vexDoc *VEX) EffectiveStatement(product, vulnID string) (s *Statement) {
 	SortStatements(statements, t)
 
 	for i := len(statements) - 1; i >= 0; i-- {
-		if statements[i].Vulnerability.ID != vulnID {
+		if !statements[i].Vulnerability.Matches(vulnID) {
```

Comment:
Instead of checking here for the vuln and then the products, we can replace this while block with `statement.Matches()`:

https://github.com/openvex/go-vex/blob/76b7c3d0be95a3ee2febf40252371bf65f381130/pkg/vex/statement.go#L176-L178

That function takes subcomponents, but you can pass an empty slice and it will match only on the product.

In the future, we need another function that also takes subcomponents for a more precise match.

Reply:
Thanks for your review. Fixed in https://github.com/openvex/go-vex/pull/54/commits/6377b4f462c0de5b584ea069ff6fcf6cfc4f4306.